### PR TITLE
refactor: use react-native-quick-crypto, remove rn-nodify

### PR DESCRIPTION
**Pull request by pwltr**
### Description

Replaces `crypto` calls with `react-native-quick-crypto` which should offer way better performance since it's using JSI/C++. Allows us to remove `rn-nodify` and for Node core module polyfilling. Also replaces `text-encoding` with [faster](https://github.com/anonyco/FastestSmallestTextEncoderDecoder#benchmarks) alternative `fastestsmallesttextencoderdecoder`. Renames `.node-version` file so it can be picked up by version managers.

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Refactoring (improving code without creating new functionality)

### Tests

- [ ] Detox test
- [ ] Unit test
- [x] No test

### QA Notes

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce during the Bitkit testing session. You can also leave a video of the PR in action.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -30,11 +29,9 @@
     "prepare": "husky install",
     "postinstall": "node postinstall.js"
   },
-  "engines": {
-    "node": ">=16"
-  },
   "dependencies": {
     "@bitcoinerlab/secp256k1": "1.0.5",
+    "@craftzdog/react-native-buffer": "^6.0.5",
```
Comment (limpbrains) [inline package.json]: please exact versions in dependencies

**Review by limpbrains — CHANGES_REQUESTED** [review]

File: ios/Podfile.lock [hunk 2]
```diff
@@ -76,9 +76,9 @@ PODS:
   - lottie-react-native (5.1.6):
     - lottie-ios (~> 3.4.0)
     - React-Core
-  - MMKV (1.3.1):
-    - MMKVCore (~> 1.3.1)
-  - MMKVCore (1.3.1)
+  - MMKV (1.3.0):
+    - MMKVCore (~> 1.3.0)
+  - MMKVCore (1.3.0)
```
Comment (limpbrains) [inline ios/Podfile.lock]: I'm qurious why MMKV version changed? I don't have it on my local maching when I do `pod install` maybe you should try `pod install --repo-update` ?

File: package.json (hunk 1 above)
Comment (pwltr) [reply package.json]: done

File: ios/Podfile.lock (hunk 2 above)
Comment (pwltr) [reply ios/Podfile.lock]: done, but now it updated it to newer versions. Not sure why it always downgrades some packages on my machine.

**Review by limpbrains — APPROVED** [review]

**Comment by coreyphillips** [comment]
Seems to break being able to set custom Electrum servers in Settings->Advanced->Electrum Server

**Review by coreyphillips — APPROVED** [review]

**Pull request merged**